tools/toolchains/iar.py

```diff
@@ -126,17 +126,22 @@ def cc_extra(self, object):
         base, _ = splitext(object)
         return ["-l", base + '.s.txt']
 
-    def get_compile_options(self, defines, includes):
+    def get_compile_options(self, defines, includes, for_asm=False):
         opts = ['-D%s' % d for d in defines] + ['-f', self.get_inc_file(includes)]
-        config_header = self.get_config_header()
-        if config_header is not None:
-            opts = opts + ['--preinclude', config_header]
+        if for_asm:
+            # The assembler doesn't support '--preinclude', so we need to add
+            # the macros directly
+            opts = opts + ['-D%s' % d for d in self.get_config_macros()]
```

Comment:
@screamerbg: with #1975 in place, uVision is not going to be an issue. For IAR, I just opened Embedded Workbench and added a definition `DUMMY_MACRO={1,2,3,4,5}` to the C/C++ Copmiler/Preprocessor tab. EW didn't complain about that, it failed with a different error regarding duplicate object files (which is the same error that I get if I remove the definition of `DUMMY_MACRO`).